src/extensions/electrodegeometry/ElectrodeGeometryWidget/ElectrodeGeometryWidget.tsx

```diff
@@ -150,6 +152,11 @@ const onUpdateLayerProps = (layer: CanvasWidgetLayer<ElectrodeLayerProps, Electr
         scaleFactor = H / getHeight(boundingBox)
     }
 
+    // We don't want to have big huge electrode circles if there are too few electrodes relative to the canvas.
+    // To correct for this, adjust the scale factor downwards if it would result in an unacceptably large radius.
+    const MAX_RADIUS_PIXELS = 32
```

Comment:
This constant could be defined elsewhere (at the least, higher up in the function) but for now I wanted to keep it close to where it was used so its function was more obvious.